pkg/k8sutil/status.go

```diff
@@ -254,3 +256,92 @@ func UpdateRollingUpgradeState(c client.Client, cluster *v1beta1.KafkaCluster, t
 	logger.Info("Rolling upgrade status updated", "status", timeStamp)
 	return nil
 }
+
+func UpdateListenerStatuses(c client.Client, cluster *v1beta1.KafkaCluster, logger logr.Logger, intListenerStatuses, extListenerStatuses map[string]v1beta1.ListenerStatusList) error {
+	typeMeta := cluster.TypeMeta
+
+	cluster.Status.ListenerStatuses = v1beta1.ListenerStatuses{
+		InternalListeners: intListenerStatuses,
+		ExternalListeners: extListenerStatuses,
+	}
+
+	err := c.Status().Update(context.Background(), cluster)
+	if apierrors.IsNotFound(err) {
+		err = c.Update(context.Background(), cluster)
+	}
+	if err != nil {
+		if !apierrors.IsConflict(err) {
+			return errors.WrapIf(err, "could not update listener status")
+		}
+		err := c.Get(context.TODO(), types.NamespacedName{
```

Comment:
```
  err := c.Get(context.TODO(), types.NamespacedName{
```

Rather create a 
```
ctx := context.Background()
```
at the beginning of the methods and use `ctx` for all the calls that requires a context. Also, the context could be injected as arg by the caller.